pilot/pkg/model/mesh.go

```diff
@@ -0,0 +1,785 @@
+	// UID.
+	UID destinationAttribute = "destination.uid"
+
+	// SERVICE represents the fully qualified __CANONOCAL__ name of the Istio
+	// service, ex: "my-svc.my-namespace.svc.cluster.local".  Mesh uses
+	// this label to identify endpoints belonging to the same service for XDS queries
+	// involving service names. The attribute value is expected to be mesh unique,
+	// meaning services having this name are semantically understood to be the same
+	// modulo version and other labels, irrespective of the platform under which
+	// they are run, the clusters or availability zones they run in.
+	SERVICE destinationAttribute = "destination.service"
+
+	// NAME represents the short part of the service name, ex: "my-svc".
+	NAME destinationAttribute = "destination.name"
+
+	// NAMESPACE represents the namespace part of the destination service,
+	// example: "my-namespace".
+	NAMESPACE destinationAttribute = "destination.namespace"
+
+	// IP represents the IP address of the server instance, example 10.0.0.104.
+	// This IP is expected to be reachable from __THIS__ pilot. No distinction
+	// is being made for directly reachable service instances versus those
+	// behind a VIP. Istio's health discovery service will ensure that this
+	// endpoint's capacity is correctly reported accordingly.
+	IP destinationAttribute = "destination.ip"
+
+	// PORT represents the recipient port on the server IP address, Example: 443
+	PORT destinationAttribute = "destination.port"
+
+	// DOMAIN represents the domain portion of the service name, excluding
+	// the name and namespace, example: svc.cluster.local
+	DOMAIN destinationAttribute = "destination.domain"
+
+	// USER represents the __CANONICAL__ Mesh user running the destination
+	// application, example: service-account-foo. The user's identity is
+	// assumed to be trusted within the entire mesh, irrespective of which
+	// platform, cluster or availability zone this instance runs in.
+	USER destinationAttribute = "destination.user"
+)
+
+var (
+	destinationAttrSet = map[string]bool{
+		UID.stringValue():       true,
+		SERVICE.stringValue():   true,
+		NAME.stringValue():      true,
+		NAMESPACE.stringValue(): true,
+		IP.stringValue():        true,
+		PORT.stringValue():      true,
+		DOMAIN.stringValue():    true,
+		USER.stringValue():      true,
+	}
+
+	canonicalIstioDomain = "svc.cluster.local"
+	svcRegexp            = regexp.MustCompile(
+		"([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.svc\\.cluster\\.local")
+	domainRegexp = regexp.MustCompile(
+		"([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])*")
+)
+
+type destinationAttribute string
+
+// Mesh is a platform independent abstraction used by Controllers
+// for maintaining a list of service endpoints used by this Pilot.
+//
+// Controllers under https://github.com/istio/istio/tree/master/pilot/platform
```

Comment:
That changed - registries...

Reply:
I'd be happy to change it. The interface is still model.Controller